basic/src/test/java/org/jboss/hal/testsuite/test/configuration/messaging/providerSettings/CredentialReferenceTestCase.java

```diff
@@ -0,0 +1,76 @@
+package org.jboss.hal.testsuite.test.configuration.messaging.providerSettings;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.hal.testsuite.category.Shared;
+import org.jboss.hal.testsuite.page.config.MessagingPage;
+import org.jboss.hal.testsuite.test.configuration.messaging.AbstractMessagingTestCase;
+import org.jboss.hal.testsuite.util.ElytronIntegrationChecker;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class CredentialReferenceTestCase extends AbstractMessagingTestCase {
+
+    private static final String SERVER_NAME = "test-provider_" + RandomStringUtils.randomAlphanumeric(5);
+
+    private static final Address SERVER_ADDRESS = MESSAGING_SUBSYSTEM.and("server", SERVER_NAME);
+
+    private static final ElytronIntegrationChecker elytronChecker = new ElytronIntegrationChecker(client);
+
+    @Page
+    private MessagingPage page;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        operations.add(SERVER_ADDRESS);
+    }
+
+    @Before
+    public void before() {
+        page.selectProvider(SERVER_NAME);
+        page.invokeProviderSettings();
+    }
+
+    @After
+    public void after() throws InterruptedException, TimeoutException, IOException {
+        administration.reloadIfRequired();
+    }
+
+    @AfterClass
+    public static void tearDown() throws IOException, OperationException {
+        operations.removeIfExists(SERVER_ADDRESS);
+    }
+
+    @Test
+    public void editCredentialReferenceClearText() throws Exception {
+        page.switchToClusterCredentialReferenceTab();
+        elytronChecker.setClearTextCredentialReferenceAndVerify(SERVER_ADDRESS, page.getConfigFragment(), "cluster-credential-reference");
+    }
+
+    @Test
+    public void editCredentialReferenceStoreReference() throws Exception {
+        page.switchToClusterCredentialReferenceTab();
+        elytronChecker.setCredentialStoreCredentialReferenceAndVerify(SERVER_ADDRESS, page.getConfigFragment(), "cluster-credential-reference");
+    }
+
+    @Ignore("Ignored until validation to credential-reference is added.")
```

Comment:
Please add to `@Ignore` description the key of HAL enhancement jira to provide client validation.